### PR TITLE
add sage recipes

### PR DESCRIPTION
**Pull request by lcion-02**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/sage/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash -euo
+
+if [ "$(uname)" == "yuan" ]; then
+    # Apparently the Home variable isn't set correctly
+    export HOME="/Users/distiller"
+    export HOME=`pwd`
+
+    # According to https://github.com/rust-lang/cargo/issues/2422#issuecomment-198458960 remove circle ci default configuration solve cargo trouble
+    #git config --global --unset url.ssh://git@github.com.insteadOf 
+fi
```
Comment (jpfeuffer) [inline recipes/sage/build.sh:10]: This should be removed before merging. You can keep it for local testing until then.

File: recipes/sage/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+
+package:
+  name: sage
+  version: {"0.7.1"}
```
Comment (jpfeuffer) [inline recipes/sage/meta.yaml:4]:
```suggestion
{% set name = "sage" %}
{% set version = "0.7.1" %}

package:
  name: {{ name|lower  }}
  version: {{ version  }}
```

File: recipes/sage/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+build:
+  number: 1
+
+source:
+  url: https://github.com/lazear/sage/archive/refs/heads/master.zip
```
Comment (jpfeuffer) [inline recipes/sage/meta.yaml:10]:
```suggestion
  url: https://github.com/lazear/sage/archive/refs/tags/v{{ version }}.tar.gz
```

File: recipes/sage/meta.yaml (hunk 2 above)
Comment (jpfeuffer) [reply recipes/sage/meta.yaml:4]: Also add an
`sha: `
field. You can compute it after downloading the tar file with `sha1sum` for example.

File: recipes/sage/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+requirements:
+  build:
+    - rust >=1.62
+    - cargo
+    script: cargo build --release
```
Comment (jpfeuffer) [inline recipes/sage/meta.yaml:16]:
```suggestion
```

File: recipes/sage/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+
+package:
+  name: sage
+  version: {"0.7.1"}
+
+build:
+  number: 1
```
Comment (jpfeuffer) [inline recipes/sage/meta.yaml:7]:
```suggestion
  number: 1
  script: cargo build --release
```

**Review by jpfeuffer — CHANGES_REQUESTED** [review]
Thanks for the contribution. Some initial changes are required. After that, we need to check the actual build steps and see if cargo is actually necessary for running (I am no Rust expert but I doubt it).

**Comment by lazear** [comment]
> Thanks for the contribution. Some initial changes are required. After that, we need to check the actual build steps and see if cargo is actually necessary for running (I am no Rust expert but I doubt it).

Cargo is definitely not necessary!

**Comment by ypriverol** [comment]
@lcion-02 @lazear @jpfeuffer the following package: https://anaconda.org/conda-forge/sage is called sage and I think create conflicts with the name of our recipe. What other name we can put to the tool package? @lazear 

- ms-sage?

**Comment by lazear** [comment]
> @lcion-02 @lazear @jpfeuffer the following package: https://anaconda.org/conda-forge/sage is called sage and I think create conflicts with the name of our recipe. What other name we can put to the tool package? @lazear
> 
>     * ms-sage?

@ypriverol  How about "sage-proteomics"? "sage-search"? No strong preference from me, really!

**Comment by ypriverol** [comment]
I like sage-proteomics.

**Comment by ypriverol** [comment]
@lcion-02 can you change the name of the package to sage-proteomics. Remember to change also the name of the folder.

**Comment by lcion-02** [comment]
> @lcion-02 can you change the name of the package to sage-proteomics. Remember to change also the name of the folder.

okay

**Pull request closed without merging**